src/plugins/BuildServerIntegration/AzureDevOpsIntegration/ApiClient.cs

```diff
@@ -133,9 +136,14 @@ private static string QueryForBuildStatus(string buildDefinitionsToQuery, string
 
         public void Dispose()
         {
-            _httpClient?.Dispose();
+            Dispose(true);
             GC.SuppressFinalize(this);
         }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            _httpClient?.Dispose();
```

Comment:
A general guidance, from what I read, is to not create-dispose many `HttpClient`s, instead use of for the life of an app. Or use `HttpClientFactory`. Otherwise, there's a risk of exhausting sockets, and generally affect the performance of a workstation.

How often is this one created?

Reply:
Agreed but the httpclient is created/disposed only when opening a new repo (because is configured with authorization settings) and it seems good enough, even if not perfect, to avoid having the port exhaustion problem.